01-data-structures/03-linked-lists/linked-lists-answers.txt

```diff
@@ -0,0 +1,25 @@
+1) Spatial Locality is when data references are found consecutively on memory. The benefit of using datastructures with spatial locality is that since the data within the datastructure is all located near each other, and RAM lookups are also done across a range of memory locations, once part of the datastructure is found the rest of the datastructure can be easily found by looking at consecutive memory locations, limiting the number of RAM lookups.
+
+2) Using Benchmark, using arrays or lists of 100000 items, the following was the output (slightly formatted for easier interpretation):
+
+              user     system      total        real
+Linitialize  0.020000   0.010000   0.030000 (  0.021833)
+Ainitialize  0.000000   0.000000   0.000000 (  0.004854)
+
+        user     system      total        real
+Lfind  0.010000   0.000000   0.010000 (  0.002413)
+Afind  0.000000   0.000000   0.000000 (  0.000001)
+
+          user     system      total        real
+Ldelete  0.000000   0.000000   0.000000 (  0.001915)
```

Comment:
This is a copy paste from some output of the Benchmark file.

linked_list_benchmark.rb